src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceTransformer.java

```diff
@@ -261,4 +277,87 @@ private GrpcDocView generateEnumView(
+
+  private ViewModel generateTypesView(Model model, GapicProductConfig productConfig) {
+    SurfaceNamer namer = new PythonSurfaceNamer(productConfig.getPackageName());
+    ImportSectionView imports =
+        importSectionTransformer.generateTypesImportSection(model, productConfig);
+    return VersionIndexView.newBuilder()
+        .templateFileName(TYPES_TEMPLEATE_FILENAME)
+        .outputPath(typesOutputFile(namer))
+        .requireViews(ImmutableList.<VersionIndexRequireView>of())
+        .apiVersion(namer.getApiWrapperModuleVersion())
+        .namespace(namer.getVersionedDirectoryNamespace())
+        .packageVersion(packageConfig.generatedPackageVersionBound(TargetLanguage.PYTHON).lower())
+        .fileHeader(fileHeaderTransformer.generateFileHeader(productConfig, imports, namer))
+        .build();
+  }
+
+  private ViewModel generateVersionedInitView(Model model, GapicProductConfig productConfig) {
+    SurfaceNamer namer = new PythonSurfaceNamer(productConfig.getPackageName());
+    boolean packageHasEnums = false;
+    for (TypeRef type : model.getSymbolTable().getDeclaredTypes()) {
+      if (type.isEnum() && type.getEnumType().isReachable()) {
+        packageHasEnums = true;
+        break;
+      }
+    }
+    ImportSectionView imports =
+        importSectionTransformer.generateVersionedInitImportSection(
+            model, productConfig, packageConfig, namer, packageHasEnums);
+    return VersionIndexView.newBuilder()
+        .templateFileName(VERSIONED_INIT_TEMPLATE_FILENAME)
+        .outputPath(versionedInitOutputFile(namer))
+        .requireViews(versionedInitRequireViews(model, productConfig, namer))
+        .apiVersion(namer.getApiWrapperModuleVersion())
+        .namespace(namer.getVersionedDirectoryNamespace())
+        .packageVersion(packageConfig.generatedPackageVersionBound(TargetLanguage.PYTHON).lower())
+        .fileHeader(fileHeaderTransformer.generateFileHeader(productConfig, imports, namer))
+        .packageHasEnums(packageHasEnums)
+        .build();
+  }
+
+  private List<VersionIndexRequireView> versionedInitRequireViews(
+      Model model, GapicProductConfig productConfig, SurfaceNamer namer) {
+    ImmutableList.Builder<VersionIndexRequireView> views = ImmutableList.builder();
+    Iterable<Interface> apiInterfaces = new InterfaceView().getElementIterable(model);
+    ModelTypeTable typeTable = emptyTypeTable(productConfig);
+    for (Interface apiInterface : apiInterfaces) {
+      views.add(
+          VersionIndexRequireView.newBuilder()
+              .clientName(namer.getAndSaveNicknameForGrpcClientTypeName(typeTable, apiInterface))
+              .localName(
+                  namer.getApiWrapperClassName(productConfig.getInterfaceConfig(apiInterface)))
+              .fileName(namer.getNotImplementedString("VersionIndexRequireView.fileName"))
+              .build());
+    }
+    return views.build();
+  }
+
+  private ModelTypeTable emptyTypeTable(GapicProductConfig productConfig) {
```

Comment:
The context which this method is being used, there is no typetable to call cloneEmpty from.